src/codecs/bmp/decoder.rs

```diff
@@ -1467,7 +1467,7 @@ impl<'a, R: 'a + Read + Seek> ImageDecoderExt<'a> for BmpDecoder<R> {
         progress_callback: F,
     ) -> ImageResult<()> {
         let start = self.reader.seek(SeekFrom::Current(0))?;
-        image::load_rect(x, y, width, height, buf, progress_callback, self, |_, _| unreachable!(),
+        image::load_rect(x, y, width, height, buf, progress_callback, self, |_, _| Ok(()),
                          |s, buf| s.read_image_data(buf))?;
         self.reader.seek(SeekFrom::Start(start))?;
```

Comment:
Hm, if `load_rect` wants to seek back to the start then I wonder if the callback might perform that relative seek, and these two lines be merged?

Reply:
And wait a minute, does this work properly when you try to read any rect other than `x = 0`?

Reply:
The trick is that there's only 1 scanline total, so load_rect just skips the part of the image that isn't required

Reply:
But what if it doesn't start at offset 0?

Reply:
Oh, I see what you're saying. What the code is _meant_ to do is read the scanline once and then copy it into the output buffer row by row. What it actually does is repeatedly read the same scanline (seeking first if necessary) and then copy one row out of it each time.

This part of the code could use significantly more testing...